src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -30,6 +30,7 @@ namespace eicrecon {
 
 void CalorimeterHitDigi::init(const dd4hep::Detector* detector, std::shared_ptr<spdlog::logger>& logger) {
     m_detector = detector;
+    m_converter = std::make_shared<const dd4hep::rec::CellIDPositionConverter>(const_cast<dd4hep::Detector&>(*detector));
```

Comment:
You will have to pass the converter from the factory to the algorithm. We don't allow creating one in the algorithms anymore. We also don't allow const_cast.